wpilibj/src/main/java/edu/wpi/first/wpilibj/apriltag/AprilTagFieldLayout.java

```diff
@@ -151,6 +153,20 @@ public void serialize(Path path) throws IOException {
     new ObjectMapper().writeValue(path.toFile(), this);
   }
 
+  /**
+   * Deserializes a field layout from a resource within a jar file.
+   *
+   * @param resourcePath The absolute path of the resource
+   * @return The deserialized layout
+   * @throws IOException If the resource could not be loaded
+   */
+  public static AprilTagFieldLayout loadFromResource(String resourcePath) throws IOException {
```

Comment:
If this and the predefined enum end up living together, I would also make a `loadFromField` function like I did for the field images